internal/exec/stages/disks/luks.go

```diff
@@ -531,3 +559,87 @@ func (d LuksDump) hasFlag(flag string) bool {
 	}
 	return false
 }
+
+// collect the cex card domains xx.xxxx,
+func cexCardDomain() (string, error) {
+	var ret []string
+	domain, err := exec.Command(distro.LszcryptCmd(), "-V", "--ccaonly", "--queueonly").CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("lszcrypt: No crypto card devices found: %w", err)
+	}
+	// Skip the unwanted lines before the actual value
+	col := strings.Split(string(domain), "\n")[2:]
+	// Pattern to match the first column to grep one or more domains without card number.
+	for _, line := range col {
+		match := cexRegx.FindStringSubmatch(line)
+		if match != nil {
+			ret = append(ret, match[1])
+		}
+	}
+	return strings.Join(ret, ","), nil
+}
+
+// Arguments specific to zkey generation
+func zkeySecGenArgs(luks types.Luks) []string {
+	ret := []string{
+		"generate",
+		"--name",
+		"root_xts",
+		"--xts",
+		"--volumes",
+		execUtil.DeviceAlias(*luks.Device),
+		"--description",
+		"root Masterkey",
+	}
+	// Verify the Keytype for securekey
+	if !util.NilOrEmpty(luks.Cexmode.Keytype) {
+		ret = append(ret, "--key-type", *luks.Cexmode.Keytype)
+	} else {
+		ret = append(ret, "--key-type", "CCA-AESCIPHER")
+	}
+	// Sector size are optional, if not provided 512
+	if luks.Cexmode.Sectorsize != nil {
+		ret = append(ret, "--sector-size", "4096")
+	}
+	return ret
+}
+
+// Secure key Generation using zkey.
+func zkeySecKeyGen(luks types.Luks) error {
+	args := zkeySecGenArgs(luks)
+	dom, err := cexCardDomain()
+	if err != nil {
+		return fmt.Errorf("Error: %w", err)
+	}
+	args = append(args, "--apqns", dom)
+	_, err = exec.Command(distro.ZkeyCmd(), args...).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("Error generating securekey: %w", err)
+	}
+	return nil
+}
+
+// zkey Cryptsetup args generate pbkdf algo as argon2i.
+// fips does not support argon2i. hence adding the
+// args manually by verifying the keysize and keytype.
+func zkeySecCryptGenArg(luks types.Luks) []string {
+	ret := []string{
+		"--master-key-file",
```

Comment:
Looks like that name is deprecated and it's now `--volume-key-file`.

Reply:
Agreed.